store/store.go

```diff
@@ -709,6 +709,16 @@ func (s *Store) retryRequestDecodeJSON(ctx context.Context, reqOptions *requestO
 
 // doRequest does an authenticated request to the store handling a potential macaroon refresh required if needed
 func (s *Store) doRequest(ctx context.Context, client *http.Client, reqOptions *requestOptions, user *auth.UserState) (*http.Response, error) {
+	// this gets used in retryRequestDecodeJSON, meaning we'll quickly retry
+	// this a few times. a potential solution is extending httputil.RetryRequest
+	// to short circuit on a specific set of errors?
+	//
+	// httputil.ShouldRetryError already does something like this. maybe add to
+	// that function a check for store.ErrStoreOffline?
```

Comment:
That can't actually be done, at least if the error lives in `store` because it would cause a cyclical import.

Reply:
mmh,  doesn't httputil.ShouldRetryError(store.ErrStoreOffline) return false anyway at this point? that's at least the default